### PR TITLE
Silence BPF firewall warning when not useful

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/core/bpf-firewall.c [hunk 1]
```diff
@@ -766,3 +764,15 @@ int bpf_firewall_supported(void) {
                 return supported = BPF_FIREWALL_UNSUPPORTED;
         }
 }
+
+void emit_bpf_firewall_warning(Unit *u) {
+        static bool warned = false;
+
+        if (!warned) {
+                log_unit_warning(u, "unit configures an IP firewall, but %s.\n"
```
Comment (yuwata) [inline src/core/bpf-firewall.c]: `unit configures` -> `Unit configures`?

File: src/core/bpf-firewall.c [hunk 2]
```diff
@@ -769,10 +781,13 @@ void emit_bpf_firewall_warning(Unit *u) {
...
 
         if (!warned) {
-                log_unit_warning(u, "unit configures an IP firewall, but %s.\n"
-                                    "(This warning is only shown for the first unit using IP firewalling.)",
-                                 getuid() != 0 ? "not running as root" :
-                                                 "the local system does not support BPF/cgroup firewalling");
+                bool quiet = bpf_firewall_unsupported_reason == -EPERM && detect_container();
+
+                log_unit_full(u, quiet ? LOG_DEBUG : LOG_WARNING, bpf_firewall_unsupported_reason,
+                              "unit configures an IP firewall, but %s.\n"
```
Comment (yuwata) [inline src/core/bpf-firewall.c:787]: `unit` -> `Unit`?

File: src/core/bpf-firewall.c (hunk 1 above)
Comment (keszybz) [reply src/core/bpf-firewall.c]: `log_unit_warning` adds a prefix, so capitalized "Unit" looks wrong.

File: src/core/bpf-firewall.c (hunk 2 above)
Comment (keszybz) [reply src/core/bpf-firewall.c:787]: Same here.

File: src/core/bpf-firewall.c (hunk 1 above)
Comment (yuwata) [reply src/core/bpf-firewall.c]: OK.

**Comment by yuwata** [comment]
```
======================================================================
FAIL: test_dhcp_client_vrf (__main__.NetworkdDHCPClientTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test/test-network/systemd-networkd-tests.py", line 2291, in test_dhcp_client_vrf
    self.assertEqual(output, output_ip_vrf)
AssertionError: '75: [488 chars]t 3594sec preferred_lft 3594sec\n    inet6 fe8[81 chars]ever' != '75: [488 chars]t 3595sec preferred_lft 3595sec\n    inet6 fe8[81 chars]ever'
Diff is 776 characters long. Set self.maxDiff to None to see it.
```

**Comment by yuwata** [comment]
LGTM. CI failure should be fixed by #12744.

**Pull request merged**